demo_site/webauthn-rs-demo/src/actors.rs

```diff
@@ -50,7 +50,7 @@ impl WebauthnActor {
         let rp_name = rp_name.to_string();
         let rp_id = rp_id.to_string();
         let rp_origin = Url::parse(rp_origin).expect("Failed to parse origin");
-        let wan = unsafe { WebauthnCore::new(&rp_name, &rp_id, &rp_origin, None, None) };
```

Comment:
This needs to stay, Webauthn *is* unsafe, because it's trivial to fuck-it-up with security consequences.

Reply:
I understand the intention here, but the `unsafe` keyword is a Rust protection not application-logic "construction is safe" protection. Since there is no unsafe rust code here, it truly should not be unsafe. I.e. there's no memory unsafe operations here.

Reply:
we can change the name of the `new` constructor, but I think `unsafe` here will cause strange downstream effects. For instance, there are audits that check how much "unsafe" code exist in dependencies.

Reply:
That's also kind of the motivation, is that this *should* turn up in an audit, so that people question if it's being used correctly. But I can certainly see how from a "pure memory safety" perspective it's not "accurate". That's why  Ithink this should be a seperate issue to discuss (I'm not saying no, but I think seperate issue to focus on that one thing)

Reply:
it's tough because I think library users will want to customize there own use of WebauthnCore. Most libraries allow full customization of this. I think the safer interfaces are nice, but I needed more customization in my use case.

Reply:
Yeah, I'm wondering what your customisations are that you need? It may not be "unique" to you :)